packages/vite/src/node/plugins/css.ts

```diff
@@ -228,7 +228,7 @@ const cssModuleRE = new RegExp(`\\.module${CSS_LANGS_RE.source}`)
 const directRequestRE = /[?&]direct\b/
 const htmlProxyRE = /[?&]html-proxy\b/
 const htmlProxyIndexRE = /&index=(\d+)/
-const commonjsProxyRE = /\?commonjs-proxy/
+const commonjsProxyRE = /[?&]commonjs-proxy/
```

Comment:
I noticed another possible bug:
Vite built-in plugins use a convention of appending query-like arguments to import specifiers to update the plugin behavior (example: `?url`)
Custom plugins may also adapt this convention.

However, if `?` character is present in the import specifier, some parts of vite code incorrectly consider it to be a glob pattern:

https://github.com/vitejs/vite/blob/2c4a0920ec7730e61508745fb8e4d49fc2945f00/packages/vite/src/node/server/warmup.ts#L79
https://github.com/vitejs/vite/blob/2c4a0920ec7730e61508745fb8e4d49fc2945f00/packages/vite/src/node/optimizer/index.ts#L875
https://github.com/vitejs/vite/blob/2c4a0920ec7730e61508745fb8e4d49fc2945f00/packages/vite/src/node/optimizer/scan.ts#L341


I will refactor my plugin to not use the `?` convention

Reply:
> However, if `?` character is present in the import specifier, some parts of vite code incorrectly consider it to be a glob pattern:

All these codes receives globs (including normal file paths) instead of a list of ids (import specifiers). So it should be fine to use `?`s.